python/ray/experimental/state/common.py

```diff
@@ -561,7 +594,11 @@ class TaskState(StateSchema):
     #: The time when the task is created. A Unix timestamp in ms.
     creation_time_ms: Optional[int] = state_column(detail=True, filterable=False)
     #: The time when the task starts to run. A Unix timestamp in ms.
-    start_time_ms: Optional[int] = state_column(detail=True, filterable=False)
+    start_time_ms: Optional[int] = state_column(
+        detail=True,
+        filterable=False,
+        metadata={"format_fn": lambda x: datetime.datetime.fromtimestamp(x / 1000)},
```

Comment:
nit: why don't we just `format_fn` a standalone key-value arg, instead of embedding it in the metadata struct?

Reply:
Looks like detail and filterable fields just use metadata under the hood though? Maybe we can use the same pattern?
```

def state_column(*, filterable: bool, detail: bool = False, **kwargs):
    """A wrapper around dataclass.field to add additional metadata.

    The metadata is used to define detail / filterable option of
    each column.

    Args:
        detail: If True, the column is used when detail == True
        filterable: If True, the column can be used for filtering.
        kwargs: The same kwargs for the `dataclasses.field` function.
    """
    m = {"detail": detail, "filterable": filterable}
    if "metadata" in kwargs:
        kwargs["metadata"].update(m)
    else:
        kwargs["metadata"] = m
    return field(**kwargs)
```

Reply:
Ahh I see. When you say use the same pattern, do you mean if we explicitly include a metadata parameter, we don't specify detail=?, filterable=? as separate arguments, but rather, as additional keys? Or refactor state_column to not accept detail / filterable in general and have some default keys in metadata underneath the hood if those keys are not specified?